### DISCUSSION TITLE
Issues running over VPN

### DISCUSSION THREAD

**Discussion by davthomaspilot** [tagyoureit/nodejs-poolController#827] [App and O/S Config Questions] [answered]
I'm out of town for a month and want to keep tabs on my pool, especially REM.

I have an openvpn client running on the rpi that's running njsPC.  I can make a vpn connection to that rpi and start an ssh session using putty.  But, when I try to use a web browser to open a page at the vpn URL (192.168.57.19:5150), the settings dialog comes up, connections tab, like the server cannot be found.  I do  a "find server", and I get two 169.x addresses with port 4200.

I have another openvpn client running on a different rpi.  That rpi is also running an vnc server.   So, I can vnc connect to that rpi, then use a web browser on that rpi, to view the njsPC page using the local URL.  Strangely, it's on 4200 rather than the usual 5150.

I can "limp along" for the rest of the month doing it this way, but thought there might be a quick and easy fix so that I can just use the local browser and the vpn address.

Anybody have an idea what's going on?

**Comment by johnny2678** [comment]
Have you tried Wireguard?
`WireGuard is much faster at making connections than OpenVPN, it can complete a connection within a tenth of a second.`

I run wireguard on unraid and have no problem accessing nsjpc or any other homelab services when I'm away.

Haven't tried running wireguard directly on the Pi but looks like there are guides out there - https://pimylifeup.com/raspberry-pi-wireguard/

**Reply by rstrouse** [reply, marked as answer]
You should use the 5150 address and click the checkbox to use the proxy to njsPC.  This will allow the 4200 port to remain local and vpn to the 5150 port.  This will allow REM and njsPC to remain on the local net.  The only other way to get there is using a terminal emulation like RDP or VNC.

**Reply by CodeSlinger69** [reply]
I prefer WireGuard as well. As a bonus, its built into a lot of custom router firmware these days. You don't have to run the VPN server on the RPi - it can be on any machine in your home network. For my purposes, running WireGuard on the router is easiest for me.

**Comment by davthomaspilot** [comment]
I tried using the proxy to njsPC box, port 5150, and clicking apply.  But I get a HTTP 404.

**Comment by davthomaspilot** [comment]
Ok, probably what you meant--

On the connection dialog, specify port 4200 and use proxy.  But use the 5150 port as the remote URL.

Anyway, that works.  Thanks!

**Reply by rstrouse** [reply]
Yep that is what I meant.  dashPanel does a bit of magic and acts as a relay on the socket and http connection.
